rust/src/io_utils.rs

```diff
@@ -4,20 +4,24 @@ use crc32fast::Hasher;
 
 pub struct CountingCrcWriter<W> {
     inner: W,
-    hasher: Hasher,
+    hasher: Option<Hasher>,
     count: u64,
 }
 
 impl<W> CountingCrcWriter<W> {
-    pub fn new(inner: W) -> Self {
+    pub fn new(inner: W, calculate_crc: bool) -> Self {
         Self {
             inner,
-            hasher: Hasher::new(),
+            hasher: if calculate_crc {
+                Some(Hasher::new())
+            } else {
+                None
+            },
```

Comment:
Just a nit, but you could have a `NoopHasher` instead of an if statement so that you don't need to branch on every write.